### PR TITLE
fix #1238, adds BroadcastPartitioner

### LINKED ISSUE #1238: Support broadcast partitioner

### PR DESCRIPTION
**Pull request by manuzhang**

### CODE REVIEW

**Comment by whjiang** [comment]
+1.

File: streaming/src/main/scala/io/gearpump/streaming/task/Subscription.scala [hunk 1]
```diff
@@ -74,17 +75,29 @@ class Subscription(
     transport.transport(ackRequest, allTasks: _*)
   }
 
+  def sendMessage(msg: Message): Int = {
+    partitioner match {
+      case up: UnicastPartitioner =>
```
Comment (clockfly) [inline streaming/src/main/scala/io/gearpump/streaming/task/Subscription.scala]: Can we avoid do the pattern match every time?

Will there be possible performance degradtion?

**Comment by manuzhang** [comment]
@clockfly no pattern matching for every new message. Further comments ?

**Comment by clockfly** [comment]
+1

**Pull request merged**